app/src/main/java/com/arygm/quickfix/model/locations/NominatimLocationRepository.kt

```diff
@@ -0,0 +1,61 @@
+package com.arygm.quickfix.model.locations
+
+import java.io.IOException
+import okhttp3.*
+import org.json.JSONArray
+
+class NominatimLocationRepository(val client: OkHttpClient) : LocationRepository {
+  override fun search(
+      query: String,
+      onSuccess: (List<Location>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    val url = "https://nominatim.openstreetmap.org/search?q=${query}&format=json"
+
+    val request =
+        Request.Builder()
+            .url(url)
+            .header(
+                "User-Agent", "QuickFix/1.0 (daferyassine52@gmail.com)") // Set a proper User-Agent
+            .header("Referer", "https://QuickFix.com") // Optionally add a Referer
```

Comment:
we should check this, we do not own the QuickFix domain and putting a personnal email address may not be the best idea

Reply:
Thank you for bringing this up! You're absolutely right. Using a personal email address and referencing a domain we don’t own may not be ideal. I will consult with the team to determine the best approach. Let me know if you have any specific recommendations!